docs/flow/modin/experimental/xgboost.rst

```diff
@@ -63,11 +63,28 @@ for any own case.
 3. ``ray.util.placement_group`` is creating to block of all available Ray resources. After that objects of ``ModinXGBoostActor``
 is being created using resources of previously created placement group.
 
+4. Data (`dtrain` for ``_train``, `data` for ``_predict``) is being splitted between actors evenly. Internal function
+``_split_data_across_actors`` runs assigning of row partitions to actors using internal function ``_assign_row_partitions_to_actors``.
+This function creates dictionary in the form: `{actor_rank: ([part_i0, part_i3, ..], [0, 3, ..]), ..}` - for training,
+`{actor_rank: [part_i0, part_i1, ..], ..}` - for prediction.
 
+**Note**: ``_assign_row_partitions_to_actors`` considers IP addresses of row partitions of `dtrain` data to minimize excess 
+data transfer.
 
+5. For each actor object method ``ModinXGBoostActor.set_train_data``/``ModinXGBoostActor.set_predict_data`` is being called remotelly to
+run loading row partitions belonging to actor according to dictionary from previous step. When data is passed in actor, auto-materialization
+of row partitions (``ray.ObjectRef`` -> ``pandas.DataFrame``) happens.
 
+6. Method ``ModinXGBoostActor.train``/``ModinXGBoostActor.predict`` is being called remotelly.
 
+* `train`: method runs xgboost training on local data of actor, connects to ``Rabit Tracker`` for sharing training state between
+  actors and returns dictionary with `booster` and `evaluation results`.
+* `predict`: method runs xgboost prediction on local data of actor and returns IP address of actor and partial prediction (``pandas.DataFrame``).
 
+7. On the final stage getting of results from actors is happened.
 
-
-
+* `train`: using ``ray.get`` function it's returned `booster` and `evals_result` from remote actor. Placement group which was
```

Comment:
Maybe "`booster` and `evals_result`  are returned from remote actor"

Reply:
fixed